crawler/file_processing.py

```diff
@@ -750,6 +780,23 @@ def get_channel_headers_mapping(self) -> Dict[str, str]:
         """
         return self.CHANNEL_FIELDS_MAPPING
 
+    def correct_headers(self, csvreader: DictReader) -> None:
+        """Checks for any headers in the CSV file that are wrong but recognisable, and fixes them.
+        Necessary due to variability in the file format we receive.
+        """
+        logger.debug("Checking CSV for nearly-correct header names and fixing them")
+
+        if csvreader.fieldnames:
+            for i in range(0, len(csvreader.fieldnames)):
+                for reg in self.header_regex_correction_dict.keys():
+                    match = re.match(reg, csvreader.fieldnames[i])
+                    if match:
+                        logger.warning(
+                            f"Found '{reg}' in field name '{csvreader.fieldnames[i]}', "
+                            f"correcting to '{self.header_regex_correction_dict[reg]}'"
+                        )
+                        csvreader.fieldnames[i] = self.header_regex_correction_dict[reg]
```

Comment:
mypy fails here with error:
`Unsupported target for indexed assignment ("Sequence[str]")`

I think this is because the type 'DictReader' is not expecting to have its items assigned (read-only).

It does feel a bit wrong to be modifying this object, what do others think?

I did it this way - a quick run through to 'correct' any headers that are a bit wrong - because that means none of the validating or parsing code needs to change - they can just assume the headers are correct.

Any thoughts welcome, thanks.

Reply:
🤔 

- what about assigning another attribute which we can check later? `corrected_fieldnames`?
- Another option, subclassing DictReader. 
- Another, adding a boolean attribute `is_fieldnames_corrected` then check `corrected_fieldnames` when needed else just `fieldnames`.

Reply:
OK so I've been thinking about this one some more. I misunderstood the error in my previous message. 

It actually means that it thinks `csvreader.fieldnames` is of the type `Sequence[str]`, and for some types of sequence you can't assign values to items in the sequence using indexing (e.g. you can't do `myString[1] = 'f'` or myTuple[1] = 'thing', but you can do `myList[1]` = 'thing')

However, it is actually of the type `list`, so indexed assignment is fine.

So now I think this is mypy's problem, and I've added a # type: ignore to it.

What do you think?